libraries/Microsoft.Bot.Builder.Alexa/Middleware/AlexaIntentRequestToMessageActivityMiddleware.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.Bot.Builder.Alexa.Middleware
+{
+    public class AlexaIntentRequestToMessageActivityMiddleware : IMiddleware
```

Comment:
>AlexaIntentRequestToMessageActivityMiddleware [](start = 17, length = 45)

I like that this is middleware, because someone may decide they want the raw intent as an activity type and so they just don't include this  piece.

Reply:
With this approach I could train my model on the pseudo text you are passing, but that would muddy up my normal model. 

My guess is that bringing this in as an raw IntentRequest means I can bypass my LuisRecognizer and just treat it as the output of a recognizer.  

Stated another way, I think perhaps there should be an IRecognizer implementation for an Alexa IntentRequest formatted object.

---
In reply to: [182599058](https://github.com/Microsoft/botbuilder-dotnet/pull/453#discussion_r182599058) [](ancestors = 182599058)

Reply:
The intent (excuse the pun) here is to provide flexibility for the user, as I think developer needs when using the Alexa adapter will vary.  The original Alexa request is always placed onto the Activity within the ChannelData object if they want access to that.  Then, as your pointed out, if you don't use the middleware then you simply get the original Alexa request type as the Activity type passed to your bot. I have also considered placing the recognized intent and slot values into the context.Services collection in the same way that the LUIS recognizer middleware does.  

Having the feature where the message text is made up of intent / slot values does mean that you can train your LUIS model to recognize intent / entities and you might choose to use this if you are also using a Google Actions adapter, for example. 

We also have two additional options. One simply takes the value of a slot called "Phrase" which can give you the entire utterance spoken by the user to Alexa, which I think will be commonly used, especially when building a bot that support Alexa and other channels like Facebook, Webchat etc.

Ultimately, if the above options (which I will document in a readme once we have landed on the final version) are not suitable then the user can pass in their own function to determine how the Message activity text should be constructed.